sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/MapStateImpl.scala

```diff
@@ -73,24 +74,31 @@ class MapStateImpl[K, V](
   }
 
   /** Get the map associated with grouping key */
-  override def getMap(): Map[K, V] = {
+  override def getMap(): Iterator[(K, V)] = {
     val encodedGroupingKey = stateTypesEncoder.encodeGroupingKey()
-    store.prefixScan(encodedGroupingKey, stateName)
-      .map {
-        case iter: UnsafeRowPair =>
-          (stateTypesEncoder.decodeCompositeKey(iter.key),
-            stateTypesEncoder.decodeValue(iter.value))
-      }.toMap
+    val pairsIterator = store.prefixScan(encodedGroupingKey, stateName)
+
+    new Iterator[(K, V)] {
```

Comment:
maybe final nit: your previous code just works and it seems much simpler. We just don't call `.toMap` and done. 

Also maybe the method name to remove the map from the name? 

* `getIterator` - consistent with getKeys/getValues
* `iterator` - consistent with Map collection. need to change other methods as well, e.g. getKeys to keys, getValues to values
* `get()` - consistent with other type of state. The value of origin type is retrieved with get() consistently.

Reply:
It's OK to defer the change of method name in follow up PR if we want to have a time to figure out the best name. The first one is something good to do this before merging the code.

Reply:
Thanks Jungtaek! As discussed, I changed `getMap()` name to `iterator` and pair it with `getKeys()`, `getValues()` to `keys()` and `values()` accordingly.